docs/overview/README.md

```diff
@@ -0,0 +1,144 @@
+hash ring with a set of tokens they own. Distributors then find the token that
+most closely mathces the value of the log's hash and will send data to that
+token's owner.
+
+#### Quorum consistency
+
+Since all distributors share access to the same hash ring, write requests can be
+sent to any distributor.
+
+To ensure consistent query results, Loki uses
+[Dynamo-style](https://www.allthingsdistributed.com/files/amazon-dynamo-sosp2007.pdf)
+quoum consistency on reads and writes. This means that the distributor will wait
+for a positive response of at least one half plus one of the ingesters to send
+the sample to before responding to the user.
+
+### Ingester
+
+The **ingester** service is responsible for writing log data to long-term
+storage backends (DynamoDB, S3, Cassandra, etc.).
+
+Logs from each unique set of labels are built up into "chunks" in memory and
+then flushed to the backing storage backend.
+
+If an ingester process crashes or exits abruptly, all the data that has not yet
+been flushed will be lost. Loki is usually configured to replicate multiple
+replicas (usually 3) of each log to mitigate this risk.
+
+#### Handoff
+
+By default, when an ingester is shutting down and tries to leave the hash ring,
+it will wait to see if a new ingester tries to enter before flushing and will
+try to initiate a handoff. The handoff will transfer all of the tokens and
+in-memory chunks owned by the leaving ingester to the new ingester.
+
+This process is used to avoid flushing all chunks when shutting down, which is a
+slow process.
+
+#### Filesystem Support
+
+While ingesters do support writing to the filesystem through BoltDB, this only
+works in single-process mode as [queriers](#querier) need access to the same
+back-end store and BoltDB only allows one process to have a lock on the DB at a
+given time.
+
+### Querier
+
+The **querier** service handles the actual [LogQL](../logql.md) evaluation of
+logs stored in long-term storage.
+
+It first tries to query all ingesters for in-memory data before falling back to
+loading data from the backend store.
+
+## Chunk Store
+
+The **chunk store** is Loki's long-term data store, designed to support
+interactive querying and sustained writing without the need for background
+maintainence tasks. It consists of:
+
+* An index for the chunks. This index can be backed by
+  [DynamoDB from Amazon Web Services](https://aws.amazon.com/dynamodb),
+  [Bigtable from Google Cloud Platform](https://cloud.google.com/bigtable), or
+  [Apache Cassandra](https://cassandra.apache.org).
+* A key-value (KV) store for the chunk data itself, which can be DynamoDB,
+  Bigtable, Cassandra again, or an object store such as
+  [Amazon * S3](https://aws.amazon.com/s3)
+
+> Unlike the other core components of Loki, the chunk store is not a separate
+> service, job, or process, but rather a library embedded in the three services
+> that need to access Loki data: the [ingester](#ingester) and [querier](#querier).
```

Comment:
and what is the third one?

Reply:
whoops :) I stole too directly from Cortex, there's only two for Loki.